src/Products/Comparison/Service/ComparisonServiceImpl.cs

```diff
@@ -92,178 +95,206 @@ public bool CheckFiles(CompareRequest files)
-                }
-                return loadDocumentEntity;
-            }
-            catch (Exception ex)
-            {
-                // set exception message
-                throw new FileLoadException("Exception occurred while loading document info", ex);
-            }
+            comparer.Source.GeneratePreview(previewOptions);
+
+            return result;
         }
 
+        private static void UserReleaseStreamMethod(int pageNumber, Stream stream)
+        {
+            stream.Close();
+        }
 
+        private static LoadOptions GetLoadOptions(string password)
+        {
+            LoadOptions loadOptions = new LoadOptions
+            {
+                Password = password
+            };
+
+            return loadOptions;
+        }
 
         private CompareResultResponse CompareTwoDocuments(CompareRequest compareRequest)
         {
             // to get correct coordinates we will compare document twice
             // this is a first comparing to get correct coordinates of the insertions and style changes
-            ICompareResult compareResult = CompareFiles(compareRequest);
             string extension = Path.GetExtension(compareRequest.guids[0].GetGuid());
+            string guid = System.Guid.NewGuid().ToString();
+            //save all results in file
+            string resultGuid = Path.Combine(globalConfiguration.Comparison.GetResultDirectory(), guid + extension);
+
+            Comparer compareResult = CompareFiles(compareRequest, resultGuid);
             ChangeInfo[] changes = compareResult.GetChanges();
-            CompareResultResponse compareResultResponse = GetCompareResultResponse(compareResult, changes, extension);
+
+            CompareResultResponse compareResultResponse = GetCompareResultResponse(changes, resultGuid);
             compareResultResponse.SetExtension(extension);
             return compareResultResponse;
         }
 
-        private ICompareResult CompareFiles(CompareRequest compareRequest)
+        private Comparer CompareFiles(CompareRequest compareRequest, string resultGuid)
         {
             string firstPath = compareRequest.guids[0].GetGuid();
-            ICompareResult compareResult;
+            string secondPath = compareRequest.guids[1].GetGuid();
+
             // create new comparer
-            Comparer comparer = new Comparer();
-            // create setting for comparing
-            ComparisonSettings settings = new ComparisonSettings();
-            settings.StyleChangeDetection = true;
-            settings.CalculateComponentCoordinates = true;
-
-            compareResult = comparer.Compare(firstPath,
-                compareRequest.guids[0].GetPassword(),
-                compareRequest.guids[1].GetGuid(),
-                compareRequest.guids[1].GetPassword(),
-                settings);
-
-            if (compareResult == null)
+            using (Comparer comparer = new Comparer(firstPath))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the 'using' statement; it will cause automatic disposal of 'comparer'.](https://app.codacy.com/manual/GroupDocs/GroupDocs.Comparison-for-.NET-MVC/pullRequest?prid=5354505)